File: All Lab Tasks/Lab_9_MediatryPattern/project/src/NextPrayerWidget.java
public class NextPrayerWidget extends WidgetComponent
{
    private String nextPrayer;

    public NextPrayerWidget(Mediator mediator) {
        super(mediator);
    }

    public void updateNextPrayer(String prayer) {
        this.nextPrayer = prayer;
        System.out.println("Next Prayer Updated: " + nextPrayer);
    }

}

File: Lab9_Mosque_Management_System/Lab9_Mosque_Management_System/Widgets/NextPrayerWidget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab9_Mosque_Management_System
{
    public class NextPrayerWidget : IWidget
    {
        private IMediator _mediator;

        public void SetMediator(IMediator mediator)
        {
            _mediator = mediator;
        }

        public void Update()
        {
            Console.WriteLine("Next Prayer updated based on Current Time and Prayer Times Table.");
        }
    }
}
